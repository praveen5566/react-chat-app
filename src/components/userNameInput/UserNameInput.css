.userNameInput-container {
    text-align: center;
}

.userNameInput-logo {
    height: 200px;
}

.userNameInput-body {
    background-color: white;
    min-height: 100vh;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    font-size: calc(10px + 2vmin);
    color: white;
}

.userNameInput-textfield {
    position: relative;
    background: white;
    color: black;
    text-transform: capitalize;
    text-decoration: none;
    border: 0.1em solid grey;
    height: 3em;
    width: 100%;
    margin: 1em;
    font-size: 0.5em;
    border-radius: 4px;
    width: 100%;
    outline: none;
    /*add outlines for 508 compliance*/
}

.userNameInput-textfield:hover {
    border: 0.1em solid var(--app-theme-color);
}


.userNameInput-button {
    position: relative;
    background: var(--app-theme-color);
    color: white;
    text-transform: capitalize;
    text-decoration: none;
    border: 0.2em solid rgb(235, 23, 0);
    padding: 1em;
    margin: 1em;
    font-size: 0.5em;
    border-radius: 4px;
    width: 100%;
    outline: none;
}

.userNameInput-button:disabled {
    background: grey;
    border: 0.2em solid grey;
}

.userNameInput-button:disabled:hover {
    cursor: not-allowed;
}

.userNameInput-button::before {
    content: "";
    display: block;
    position: absolute;
    width: 10%;
    background: white;
    height: 0.3em;
    right: 20%;
    top: -0.21em;
    transform: skewX(-45deg);
    -webkit-transition: all 0.45s cubic-bezier(0.86, 0, 0.07, 1);
    transition: all 0.45s cubic-bezier(0.86, 0, 0.07, 1);
}

.userNameInput-button::after {
    content: "";
    display: block;
    position: absolute;
    width: 10%;
    background: white;
    height: 0.3em;
    left: 20%;
    bottom: -0.25em;
    transform: skewX(45deg);
    -webkit-transition: all 0.45 cubic-bezier(0.86, 0, 0.07, 1);
    transition: all 0.45s cubic-bezier(0.86, 0, 0.07, 1);
}

.userNameInput-button:hover {
    cursor: pointer;
}

.userNameInput-button:hover::before {
    right: 80%;
}

.userNameInput-button:hover::after {
    left: 80%;
}