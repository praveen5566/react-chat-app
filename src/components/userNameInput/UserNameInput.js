import React, { useState, useEffect } from 'react';
import './UserNameInput.scss';
import { Link } from 'react-router-dom';
import { USER_NAME } from '../../constants';

export const UserNameInput = () => {
    const [userName, setUserName] = useState('');
    const [disable, setDisable] = useState(true);

    useEffect(() => {
        userName ? setDisable(false) : setDisable(true);
    }, [userName]);

    return (
        <div className="userNameInput-container">
            <div className="userNameInput-body">
                <div className="userNameInput-card fade-in-down">
                    <input className="userNameInput-textfield"
                        name="userName"
                        type="text"
                        placeholder="Type your username..."
                        maxLength="256"
                        onChange={e => setUserName(e.target.value)}
                    />
                    <Link onClick={event => (!userName) ? event.preventDefault() : null} to={`/chatWindow?${USER_NAME}=${userName}`}>
                        <button className="userNameInput-button" disabled={disable}>Join a Chat Room!</button>
                    </Link>
                </div>
            </div>
        </div>
    );
}