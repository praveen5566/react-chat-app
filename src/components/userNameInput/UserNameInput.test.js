import React from 'react';
import { shallow } from 'enzyme';
import { UserNameInput } from './UserNameInput';

it('should render proper UI', () => {
    const wrapper = shallow(<UserNameInput/>);
    expect(wrapper).toMatchSnapshot();    
});
