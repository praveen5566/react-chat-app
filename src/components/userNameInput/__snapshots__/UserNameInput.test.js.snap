// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render proper UI 1`] = `
<div
  className="userNameInput-container"
>
  <div
    className="userNameInput-body"
  >
    <img
      alt="logo"
      className="userNameInput-logo fade-in"
      src="ddLogo.svg"
    />
    <div
      className="userNameInput-card fade-in-down"
    >
      <input
        className="userNameInput-textfield"
        maxLength="256"
        name="userName"
        onChange={[Function]}
        placeholder="Type your username..."
        type="text"
      />
      <Link
        onClick={[Function]}
        to="/chatWindow?username="
      >
        <button
          className="userNameInput-button"
          disabled={true}
        >
          Join the DoorDash Chat!
        </button>
      </Link>
    </div>
  </div>
</div>
`;
